### ISSUE TITLE
Docker compose profiles set to * doesn't work

### ISSUE THREAD

**Issue by kinland** [tilt-dev/tilt#6345]
### Expected Behavior
As per the docs:
> If you want to enable all profiles at the same time, you can run docker compose --profile "*".

### Current Behavior
`docker_compose('./docker-compose.yml', profiles=["*"]` produces errors about "no Docker Compose service found with name"

### Steps to Reproduce
1. Create a docker-compose.yml with profiles
2. Create a tiltfile that has the line from the Current Behavior section
3. Run `tilt up` and see error
4. Stop tilt, then replace the wildcard profiles with the explicit list of profiles
5. Run `tilt up` and see that everything works

#### `tilt doctor` Output
```
$ tilt doctor
Tilt: vv0.33.11-next, built 2024-03-14
System: linux-amd64
---
Docker
- Host: unix:///var/run/docker.sock
- Server Version: 25.0.3
- API Version: 1.44
- Builder: 2
- Compose Version: v2.24.6-desktop.1
---
Kubernetes
- Env: eks
- Context: arn:aws:eks:us-west-2:****:cluster/****
- Cluster Name: arn:aws:eks:us-west-2:****:cluster/****
- Namespace: default
```

**Comment by nicks** [comment]
Hmmm... i was not able to reproduce this report.

from your `tilt doctor` output, it looks like you've got a custom tilt build, is it possible this is a bug specific to your branch?

**Comment by nicks** [comment]
see the circleci run on https://github.com/tilt-dev/tilt-example-docker-compose/pull/7 to see an example of this working

**Comment by kinland** [comment]
I'll try to make a reproducible example I can share next week.

**Comment by kinland** [comment]
I was on a build of your `master` branch from about a week ago. I tried switching to the latest release, and the same thing happens there. In the linked run, you're using a newer docker compose version, but I looked at their changelog since my version (`Docker Compose version v2.24.6-desktop.1`) and they make no mention of profiles.

Repro:
1. `git clone git@github.com:tilt-dev/tilt-example-docker-compose.git`
2. add profiles line to Redis:
```
    image: redis
    profiles: ['a']
```
3. add profiles to the Tiltfile: `docker_compose('docker-compose.yml', profiles=["*"])`
4. run `tilt up`

Result:
![image](https://github.com/tilt-dev/tilt/assets/16787581/845a2444-4f13-4142-8a6a-440576012ec5)

Tilt Doctor:
```
Tilt: v0.33.11, built 2024-02-15
System: linux-amd64
---
Docker
- Host: unix:///var/run/docker.sock
- Server Version: 25.0.3
- API Version: 1.44
- Builder: 2
- Compose Version: v2.24.6-desktop.1
---
Kubernetes
- Env: eks
- Context: arn:aws:eks:us-west-2:****:cluster/****
- Cluster Name: arn:aws:eks:us-west-2:****:cluster/****
- Namespace: default
- ```

**Comment by nicks** [comment]
i confirmed that the issue repro'd in Docker Compose v2.25, but is fixed in Docker Compose v2.27
https://app.circleci.com/pipelines/github/tilt-dev/tilt-example-docker-compose/28/workflows/c2962f01-23eb-4df3-ab90-78c99c346ccb/jobs/29
https://app.circleci.com/pipelines/github/tilt-dev/tilt-example-docker-compose/27/workflows/9e0b914f-5411-4e60-baec-83ee0a07276c/jobs/28

so i'm going to close this as a Docker Compose bug that's resolved now
